dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -1492,19 +1493,29 @@ def median_abs_deviation(self) -> float:
             np.append([True], np.diff(bin_edges_impose) > 1e-14)
         ]
 
-        bin_counts_impose_pos: float = np.interp(
-            bin_edges_impose, bin_edges_pos, np.cumsum(np.append([0], bin_counts_pos))
+        bin_counts_impose_pos: float = cast(
+            float,
+            np.interp(
+                bin_edges_impose,
+                bin_edges_pos,
+                np.cumsum(np.append([0], bin_counts_pos)),
+            ),
         )
-        bin_counts_impose_neg: float = np.interp(
-            bin_edges_impose, bin_edges_neg, np.cumsum(np.append([0], bin_counts_neg))
+        bin_counts_impose_neg: float = cast(
+            float,
+            np.interp(
+                bin_edges_impose,
+                bin_edges_neg,
+                np.cumsum(np.append([0], bin_counts_neg)),
+            ),
         )
         bin_counts_impose = bin_counts_impose_pos + bin_counts_impose_neg
 
         median_inds = np.abs(bin_counts_impose - 0.5) < 1e-10
         if np.sum(median_inds) > 1:
-            return np.mean(bin_edges_impose[median_inds])
+            return cast(float, np.mean(bin_edges_impose[median_inds]))
```

Comment:
I think these would also be np.float64

Reply:
Casting this to `np.float64` and changing the return type to `Union[float, np.float64]` gives me a similar issue as one of the other methods:
```
"median_absolute_deviation": utils.find_diff_of_numbers(
                self.median_abs_deviation, other_profile.median_abs_deviation
            ),
```
```
dataprofiler/profilers/numerical_column_stats.py:381: error: Value of type variable "T" of "find_diff_of_numbers" cannot be "object"
```

Reply:
Did google provide any response to this?

Reply:
so it has something to do with the return of these numpy funcs being any maybe?

Reply:
Should be fixed. See https://github.com/capitalone/DataProfiler/pull/703#discussion_r1015985771